File: AIQuranTeacherProject/android/app/src/main/java/com/example/aiquranteacher/models/Ayah.kt
package com.example.aiquranteacher.models

data class Ayah(
    val id: Int,
    val surahId: Int,
    val number: Int,
    val text: String,
    val translation: String?,
    val referenceAudioUrl: String?,
    val userRecordingUrl: String?
)

File: ai-quran-teacher/ios/AIQuranTeacher/Models/Ayah.swift
import Foundation

/// A single verse, with its mushaf text and translation.
struct Ayah: Identifiable, Codable, Hashable {
    let id: Int              // global ayah number (1...6236)
    let surahId: Int
    let numberInSurah: Int
    let text: String         // fully vocalized Arabic text
    let translation: String?
    let audioURL: URL?
    var isBookmarked: Bool = false

    enum CodingKeys: String, CodingKey {
        case id, surahId, numberInSurah, text, translation, audioURL
    }
}
